pkg/collector/check/stats/stats.go

```diff
@@ -249,6 +257,9 @@ func (cs *Stats) Add(t time.Duration, err error, warnings []error, metricStats S
 		cs.TotalEventPlatformEvents[k] = cs.TotalEventPlatformEvents[k] + v
 		cs.EventPlatformEvents[k] = v
 	}
+	if haagent != nil && haagent.Enabled() && haagent.IsHaIntegration(cs.CheckName) {
+		tlmHaAgentIntegrationRuns.Inc(cs.CheckName, pkgconfigsetup.Datadog().GetString("config_id")) // TODO: Replace with haagent.GetConfigID()
```

Comment:
Is this `TODO` something you plan to address in this PR?

Reply:
@vickenty I intended to address that later as separate PR? Is that fine for you?

Alternative, I can wait for this one to be merged https://github.com/DataDog/datadog-agent/pull/33234
Then, I will merge `main` and update this PR to use `haagent.GetConfigID()`.

Reply:
Note, the behaviour won't change, behind the scene `haagent.GetConfigID()` returns the same value as `pkgconfigsetup.Datadog().GetString("config_id")` .